x12-edi/2.5/modules/ROOT/pages/x12-edi-connector-control-numbers.adoc

```diff
@@ -0,0 +1,28 @@
+= X12 EDI Connector 2.5 - Control Numbers - Mule 4
+
+Using the Write operation in the Anypoint X12 EDI Connector, you can generate control numbers on the generated X12 payload by:
+
+* Managing global control numbers through control number keys
+* Managing identifier combination control numbers
+* Using your own control number
+
+== Manage Global Control Numbers Through Control Number Keys
+
+If your organization or your trading partner have must maintain the uniqueness of interchange, group, or transaction set control numbers across different identifier combinations and document types, then:
+
+* Set the configuration `Use supplied values` to `false`.
+* Leverage the control number key configurations (at the configuration level or operation level) to set a key unique to the partner. +
+By default, the control number key configurations pass dynamically at the operation level. You can then use this group control number key across different identifiers and transaction types for the same partner.
+
+image::X12-edi-control-number.jpg[X12 EDI Connector Control Numbers]
+
+== Manage Identifier Combination Control Numbers
+
+If you want the connector to automatically maintain control number sequencing based on the sender/receiver identifiers, and do not require global control numbers at a partner level, then:
+
+* Set the Write configuration `Use supplied values` to `false`.
+* When no value passes into the optional interchange, group, or transaction number key configurations, the sender/receiver identifier combination is the default.
+
+== Use Your Own Control Number
+
+If you must generate the control numbers and use them on the EDI payload, you can set the parameter `Use supplied values` to `true`, and map the control number on the payload to the DataWeave map.
```

Comment:
Please change the ending to "in the Dataweave map"

Reply:
Done!